### DISCUSSION TITLE
Is there an api to listen for cursor focus moving from a text editor to a terminal

### DISCUSSION THREAD

**Discussion by dddssw** [microsoft/vscode-discussions#2466] [Extension Development QnA]
I want to get the last focus position that left the text editor, because I need to click on the terminal to indicate that I want to operate the terminal

**Comment by olguzzar** [comment]
Hi @dddssw, can you provide a bit more context for your use case? Are you looking for something more than using a `when` clause for `terminalfocus`? https://code.visualstudio.com/api/references/when-clause-contexts

**Reply by dddssw** [reply]
Thank you, let me talk about the effect I want to achieve. When my mouse has a focus on the text editor, I open the terminal and enter a cli command, and then the cli will return a piece of code. I want this code to be automatically inserted into the focus at the beginning of the text editor. I don't know if this is possible
